src/redux/productsRedux.js

```diff
@@ -1,10 +1,19 @@
 /* selectors */
 export const getAll = ({ products }) => products;
 export const getCount = ({ products }) => products.length;
+export const param = ({ products }) => products.price;
 
 export const getNew = ({ products }) =>
   products.filter(item => item.newFurniture === true);
 
+export const getLimitedProducts = ({ products }, limit) => products.slice(0, limit);
+
+export const sortAscByParam = ({ products }, param) =>
+  products.sort((a, b) => a.price - b.price);
+
+export const sortDescByParam = ({ products }, param) =>
+  products.sort((a, b) => b.price - a.price);
+
```

Comment:
That's nice, but why do you always sort by price? You should allow sorting by `param`. `param` could be `price`, but could be also something different. So, do not use `price`, use `param` (e.g. `a[param]`).

Reply:
When I use `a[param]` it's not working. Maybe I don't know how this should work.

Reply:
Oh, it works. Just look at the proof ;)

https://codepen.io/Raventhein/pen/XWdjeGw

Reply:
Okay. Czy w takim razie zapis powinien być taki: `(products, param)` czy `({ products }, param)`? Sprawdziłam z tym console.log i przy zapisie z nawiasami klamrowymi nie działa. Druga rzecz jak w takim razie teraz użyć tej funkcji? Jak używałam `const sortDescByParam = ({ products }, param) => products.sort((a, b) => b.price - a.price);` to wywoływałam ją `products: sortDescByParam(state, 0),` i sortowanie działało. Teraz gdy chciałam tak zrobić pojawiają się błędy i strona nie ładuje się.

Reply:
U mnie w codepenie było od razu `products`, bo nie chciało mi się robić takiego samego obiektu jak nasz `state` ;) ale rownei dobrze, możę i tak być:

https://codepen.io/Raventhein/pen/XWdjeGw?editors=1010

Wywoływanie `products: sortDescByParam(state, 0),`, bo w ogole nie wykorzystywalas param. Moglas zamiast zera dawac nawet `abc`. To nie mialo znacznie. Bo koniec konców nigdzie `param` nie wykorzystywalas. Spójrz tylko:

`products.sort((a, b) => b.price - a.price);`

Co bys tam nie dala i tak sortowało po `price`.

Kiedy zmienisz kod na `b[param] - a[param]);`, to `param` będzie miało znacznie. Wtedy jednak musisz przekazywac w tym parametrze coś sensownego.

`sortDescByParam(state, 'price')` ma sens, bo wtedy bedziemy sortować po właściwości `price`
`sortDescByParam(state, 0)`  nie ma sensu. Bo nie mamy wlasciwosci o nazwie 0 ;)